src/storage/txn/actions/mvcc.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use tikv_kv::Snapshot;
+use txn_types::{Key, TimeStamp, Value, Write};
+
+use crate::storage::mvcc::{Lock as MvccLock, MvccReader};
+
+type LockWritesVals = (
+    Option<MvccLock>,
+    Vec<(TimeStamp, Write)>,
+    Vec<(TimeStamp, Value)>,
+);
+
+pub fn find_mvcc_infos_by_key<S: Snapshot>(
```

Comment:
What is the semantic definition or interface contract of this function? For WRITE CF, it seems to use a timestamp as an upper bound, while for DEFAULT CF, it scans all versions. Is this understanding correct?

Reply:
How about retrieving data from both CFs for all versions, but setting a maximum result size to prevent performance issues in extreme cases??

Reply:
`find_mvcc_infos_by_key` is an existing function moved from https://github.com/tikv/tikv/pull/18426/files#diff-1d92ed342cae0478fe17f515900c6aa5d3db4ad58a4eb06f743d6c9dc220094bL570. Maybe we should keep it as before in this PR to keep the existing API behavior unchanged and schedule another PR if we want to do this kind of enhancement